python/lsst/dax/ppdb/sql/ppdb_sql_utils.py

```diff
@@ -0,0 +1,333 @@
+# This file is part of dax_ppdb
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ("PpdbSqlUtils",)
+
+import logging
+import sqlite3
+from collections.abc import MutableMapping
+from contextlib import closing
+from typing import Any
+
+import felis
+import sqlalchemy
+from lsst.dax.apdb import (
+    VersionTuple,
+    schema_model,
+)
+from lsst.dax.apdb.sql import ApdbMetadataSql, ModelToSql
+from sqlalchemy.pool import NullPool
+
+from .ppdb_sql_config import PpdbSqlConfig
+
+_LOG = logging.getLogger(__name__)
+
+
+def _onSqlite3Connect(
+    dbapiConnection: sqlite3.Connection, connectionRecord: sqlalchemy.pool._ConnectionRecord
+) -> None:
+    # Enable foreign keys
+    with closing(dbapiConnection.cursor()) as cursor:
+        cursor.execute("PRAGMA foreign_keys=ON;")
+
+
+class PpdbSqlUtils:
```

Comment:
I do not understand what you are trying to do. It looks like you copied a significant piece of `PpdbSql` here. If you want to refactor things then do not duplicate the code, but make a reusable base class which you can inherit in both PpdbSql and ChunkExporter. PpdbSql is the only implementation that has unit test, try to leverage that to test your new class as well.